contribute-to-tidb/review-a-pr-stage.md

```diff
@@ -0,0 +1,238 @@
+
+A particular type of complexity is over-engineering, which means that PR authors have made the code more generic than it needs to be, or added functionality that is not presently needed by the system. Reviewers should be especially vigilant about over-engineering. The TiDB community encourages PR authors to solve the problems that must be solved now -- not the problems that might need to be solved in the future. 
+
+### Tests
+
+A PR should be test covered, whether the tests are unit tests, integration tests, or end-to-end tests.
+
+To ensure the tests in the PR are correct, sensible, and useful, you can check the following:
+
+* Whether the tests cover all interesting cases?
+* Are the test cases readable?
+* Does the PR lower the overall test coverage? 
+
+To ensure the tests in the PR are sufficient, you can simulate cases where the code may break and check the following:
+
+* Will the tests fail when the code is broken?
+* Does each test make simple and useful assertions?
+* Are the tests separated appropriately between different test methods?
+
+Remember that tests are also code that have to be maintained, and thus do not accept unnecessary complexity in tests.
+
+Last but not least, keep an eye on how long a single unit takes to run. That is, determine whether the test slow down the CI.
+
+### Naming
+
+Does the developer pick good names for everything? A good name is long enough to fully communicate what the item is or does, without being so long that it becomes hard to read.
+
+### Comments
+
+* Does the developer write clear comments in understandable English?
+* Are all of the comments actually necessary?
+
+Usually, comments are useful when they explain why the code exists, while not explain what the code is doing. If the code isn't clear enough to explain itself, then the code should be simplified.
+
+There are some exceptions; for example, regular expressions and complex algorithms often benefit greatly from comments that explain what they’re doing. But comments are mostly for information, like the reason behind a decision.
+
+It can also be helpful to look at comments existing before this PR. Maybe there is a `TODO` can be removed now, a comment advising against this change being made, etc.
+
+Note that comments are different from documentation of classes, modules, or functions, which should instead express the purpose of a piece of code, how it should be used, and how it behaves when used.
+
+### Style
+
+Make sure the PR follows our [code style and quality guide](code-style-and-quality-guide.md). For Go and Rust, there are built-in tools with the compiler toolchain.
+
+If you want to improve style points outside the style guide, you can comment that you think it would improve the code readability but isn't mandatory. Don't block PRs from being submitted based only on personal style preferences.
```

Comment:
```suggestion
If you want to improve styles not covered in the style guide, you can comment that you think it would improve the code readability but it is not mandatory. Do not block PRs from being merged based only on personal style preferences.
```